server/src/test/java/org/elasticsearch/action/admin/cluster/health/ClusterHealthResponsesTests.java

```diff
@@ -80,13 +89,60 @@ private void assertClusterHealth(ClusterHealthResponse clusterHealth) {
         assertThat(clusterHealth.getNumberOfDataNodes(), Matchers.equalTo(clusterStateHealth.getNumberOfDataNodes()));
     }
 
-    ClusterHealthResponse maybeSerialize(ClusterHealthResponse clusterHealth) throws IOException {
+    ClusterHealthResponse maybeSerialize(ClusterHealthResponse clusterHealth) {
         if (randomBoolean()) {
-            BytesStreamOutput out = new BytesStreamOutput();
-            clusterHealth.writeTo(out);
-            StreamInput in = out.bytes().streamInput();
-            clusterHealth = ClusterHealthResponse.readResponseFrom(in);
+            try {
+                BytesStreamOutput out = new BytesStreamOutput();
+                clusterHealth.writeTo(out);
+                StreamInput in = out.bytes().streamInput();
+                clusterHealth = ClusterHealthResponse.readResponseFrom(in);
+            } catch (IOException e) {
+                throw new IllegalStateException(e);
+            }
         }
         return clusterHealth;
     }
+
+    @Override
+    protected ClusterHealthResponse doParseInstance(XContentParser parser) {
+        return ClusterHealthResponse.fromXContent(parser);
+    }
+
+    @Override
+    protected ClusterHealthResponse createBlankInstance() {
+        return new ClusterHealthResponse();
+    }
+
+    @Override
+    protected ClusterHealthResponse createTestInstance() {
+        int indicesSize = randomInt(20);
+        Map<String, ClusterIndexHealth> indices = new HashMap<>(indicesSize);
+        for (int i = 0; i < indicesSize; i++) {
+            String indexName = randomAlphaOfLengthBetween(1, 5) + i;
+            indices.put(indexName, ClusterIndexHealthTests.randomIndexHealthWithShards(indexName));
+        }
+
+        ClusterStateHealth stateHealth = new ClusterStateHealth(randomInt(100), randomInt(100), randomInt(100),
+            randomInt(100), randomInt(100), randomInt(100), randomInt(100),
+            randomDoubleBetween(0d, 100d, true), randomFrom(ClusterHealthStatus.values()), indices);
+
+        return new ClusterHealthResponse(randomAlphaOfLengthBetween(1, 10), randomInt(100), randomInt(100), randomInt(100),
+            TimeValue.timeValueMillis(randomInt(10000)), randomBoolean(), stateHealth);
+    }
+
+    protected ToXContent.Params getToXContentParams() {
+        Map<String, String> map = new HashMap<>();
+        map.put("level", "shards");
+        return new ToXContent.MapParams(map);
+    }
+
+    protected boolean supportsUnknownFields() {
+        return true;
+    }
+
+    private static final Pattern SHARDS_IN_XCONTENT = Pattern.compile("^indices\\." + "\\w+" + "\\.shards$");
+
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
+        return field -> "indices".equals(field) || SHARDS_IN_XCONTENT.matcher(field).find();
```

Comment:
It can be done as manual checking if String is "indices.RANDOMINDEXNAME.shards" then return true, but I thinks regex is better. I will add explanation.